webhooks/virtualmachine/validation/virtualmachine_validator.go

```diff
@@ -2372,23 +2113,163 @@ func (v validator) validateNextPowerStateChangeTimeFormat(
 	return allErrs
 }
 
+func (v validator) validateBootOptionsWhenPoweredOn(
+	ctx *pkgctx.WebhookRequestContext,
+	vm, oldVM *vmopv1.VirtualMachine) field.ErrorList {
+
+	var allErrs field.ErrorList
+	fieldPath := field.NewPath("spec", "bootOptions")
+
+	if !reflect.DeepEqual(oldVM.Spec.BootOptions, vm.Spec.BootOptions) {
+		allErrs = append(allErrs, field.Forbidden(fieldPath, updatesNotAllowedWhenPowerOn))
+	}
+
+	return allErrs
+}
+
 func (v validator) validateBootOptions(
 	ctx *pkgctx.WebhookRequestContext,
-	vm *vmopv1.VirtualMachine) field.ErrorList {
+	vm, oldVM *vmopv1.VirtualMachine) field.ErrorList {
 
 	var allErrs field.ErrorList
+	fieldPath := field.NewPath("spec", "bootOptions")
 
-	if vm.Spec.BootOptions != nil {
-		fieldPath := field.NewPath("spec", "bootOptions")
+	bootOptions := vm.Spec.BootOptions
+	if bootOptions != nil {
+		if bootOptions.BootOrder != nil {
+			create := oldVM == nil
+			allErrs = append(allErrs, v.validateBootOrder(ctx, vm, create)...)
+		}
 
+		// Both fields are required.
 		if vm.Spec.BootOptions.BootRetryDelay != nil && vm.Spec.BootOptions.BootRetry != vmopv1.VirtualMachineBootOptionsBootRetryEnabled {
```

Comment:
Some of these inner block checks could be shorten by using the add `bootOptions` var